content/en/blog/_posts/2023-08-21-job-update-post.md

```diff
@@ -0,0 +1,207 @@
+users: [PodReplacementPolicy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated)
+and [BackoffLimitPerIndex](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3850-backoff-limits-per-index-for-indexed-jobs).
+
+## Pod Replacement Policy
+
+### What problem does this solve?
+
+By default, when a pod enters a terminating state (e.g. due to preemption or
+eviction), a replacement pod is created immediately, and both pods are running
+at the same time.
+
+This is problematic for some popular machine learning frameworks, such as
+TensorFlow and [JAX](https://jax.readthedocs.io/en/latest/), which require at most one pod running at the same time,
+for a given index (see more details in the [issue](https://github.com/kubernetes/kubernetes/issues/115844)).
+
+Creating the replacement Pod before the previous one fully terminates can also
+cause problems in clusters with scarce resources or with tight budgets. These
+resources can be difficult to obtain so pods can take a long time to find
+resources and they may only be able to find nodes until the existing pods are
+fully terminated. Further, if cluster autoscaler is enabled, the replacement
+Pods might produce undesired scale ups.
+
+### How can I use it
+
+This is an alpha feature, which you can enable by enabling the `JobPodReplacementPolicy`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) in
+your cluster.
+
+Once the feature is enabled you can use it by creating a new Job, which specifies
+`podReplacementPolicy` field as shown here:
+
+```yaml
+kind: Job
+metadata:
+  name: new
+  ...
+spec:
+  podReplacementPolicy: Failed
+  ...
+```
+
+Additionally, you can inspect the `.status.terminating` field of a Job. The value
+of the field is the number of Pods owned by the Job that are currently terminating.
+
+```shell
+kubectl get jobs/myjob -o yaml
+```
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+status:
+  terminating: 3 # three Pods are terminating and have not yet reached the Failed phase
+```
+
+This can be particularly useful for external queueing controllers, such as
+[Kueue](https://github.com/kubernetes-sigs/kueue), that would calculate the
+quota and suspend the start of a new Job until the resources are reclaimed from
+the currently terminating Job.
+
+### How can I learn more?
+
+- Read the KEP: [PodReplacementPolicy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated)
+
+## Job Backoff Limit per Index
+
+### What problem does this solve?
+
+By default, pod failures for [Indexed Jobs](/docs/concepts/workloads/controllers/job/#completion-mode)
+are counted towards the global limit of retries, represented by `.spec.backoffLimit`.
+This means, that if there is a consistently failing index, it is restarted
+repeatedly until it exhausts the limit. Once the limit is exceeded the entire
```

Comment:
I think we actually check the exceeding: https://github.com/kubernetes/kubernetes/blob/f0dcf0614036d8c3cd1c9f3b3cf8df4bb1d8e44e/pkg/controller/job/job_controller.go#L818C2-L818C21. Wdyt?

Reply:
that's an implementation detail. You "reach" the limit when you do the allowed number of recreations.

Reply:
Yeah, I was thinking "reach" conveys the same as "exceed", so it might be better to be explicit, but I have no strong preference. Seems you both lean towards "reach" I'm happy to accept the remark.